concept_r_python_cheatsheet.Rmd

```diff
@@ -0,0 +1,185 @@
+# draw graphs using python and R
+
+Shuyue Xu
+
+```{r, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+## things that need to be done before fitting models
+##### 1.check unique of the values
+##### 2.check null values, consider to drop values or adding a missing indicator
+##### 3.normalize numerical features
+##### 4.handle categorical feature using Label encoding or OneHotencoding or other...
+
+<br/> 
+
+## Continuous Variables
+we always care about asymmetry, outliers, multimodality, gaps, heaping, errors
+<br/> 
+
+### Graphs that usually used for continuous Variables:
+#### 1.Histograms: R base code:hist(x, …)
+##### 1.different bin boundaries different graphs.
+##### 2.shape: 
+               skew to left means mean < median < mode
+               skew to right means mean > median > mode
+
+#### 2.Boxplot: R base code: boxplot(x, data=),       
+##### 1.multiple boxplots usually reorder by median from high to low
+##### 2.will show outliers on the graph. outliers are 
+               1.5 x hinge or fourth spread above upper-hinge
+               1.5 x hinge or fourth spread below lower-hinge<br/> 
+
+### Ways to check normal distribution of the data
+<br/> 
+
+#### qqplot: check normal distribution 
+             qqnorm(): produces a normal QQ plot of the variable
+             qqline(): adds a reference line
+##### If the data is normally distributed, the points in the QQ-normal plot lie on a straight diagonal line.
+<br/> 
+
+#### density curve line:
+##### compare density curve line with normal curve line can also check normal distribution
+<br/> 
+
+#### Shapiro Wilk test: code:shapiro.test(x)
+##### If the p-value is smaller than alpha level, we should reject the null hypothesis that data is normally distributed and make the conclusion that data is not normally distributed.
+<br/> 
+
+## Multivariate categorical data
+#### nominal vs. ordinal, ordinal vs. discrete,...
+<br/>
+### Frequency
+#### Bar plot Basic R code: barplot(x)
+##### 1. Sort in logical order of the categories (level1, level2, lever3..)(Ordinal data)
+##### 2. Sort from highest to lowest count (Nominal data)
+<br/>
+
+#### Cleveland dot plot 
+
+#### R codes for factor data
+##### 1.don't use" "levels(x) =  ", only use "level(x)"
```

Comment:
Please fix so hashtag headers are not followed by numbers since the hashtags will create numbered subheadings